gym-unity/gym_unity/envs/__init__.py

```diff
@@ -344,6 +358,79 @@ def _check_agents(self, n_agents: int) -> None:
                 "initialization. This is not supported."
             )
 
+    def _sanitize_info(self, info: BatchedStepResult) -> BatchedStepResult:
+        self._stored_info = info  # store the new original
+        if info.n_agents() == self._n_agents:
+            return info
+        n_extra_agents = info.n_agents() - self._n_agents
+        if self._n_agents is None:
+            self._n_agents = 1
+        if n_extra_agents < 0 or n_extra_agents > 2 * self._n_agents:
+            raise UnityGymException(
+                "The number of agents in the scene does not match the expected number."
+            )
+
+        # remove the done Agents
+        indices_to_keep: List[int] = []
+        for index in range(len(info.agent_id)):
+            if not info.done[index]:
+                indices_to_keep.append(index)
+
+        # set the new AgentDone flags to True
+        for index in range(len(info.agent_id)):
+            agent_id = info.agent_id[index]
+            if agent_id not in self._stored_info.agent_id:
```

Comment:
I think this should be equivalent and O(1) instead of O(N):
```suggestion
            if not self._stored_info.contains_agent(agent_id):
```